pkg_resources/__init__.py

```diff
@@ -109,13 +112,17 @@
 )
 
 
-T = TypeVar("T")
+_T = TypeVar("_T")
+_DistributionT = TypeVar("_DistributionT", bound="Distribution")
 # Type aliases
 _NestedStr = Union[str, Iterable[Union[str, Iterable["_NestedStr"]]]]
-_InstallerType = Callable[["Requirement"], Optional["Distribution"]]
+_StrictInstallerType = Callable[["Requirement"], "_DistributionT"]
+_InstallerType = Optional[Callable[["Requirement"], Optional["Distribution"]]]
```

Comment:
It feels like we should defer adding `Optional` to the end usage.
This way the resulting type annotation would be more obvious.

```suggestion
_InstallerType = Callable[["Requirement"], Optional["Distribution"]]
```